webpack/JobWizard/JobWizard.js

```diff
@@ -18,9 +18,20 @@ export const JobWizard = () => {
 
   const setDefaults = useCallback(response => {
     const responseJob = response.data;
+    const templateValues = {};
+    const inputs = responseJob.template_inputs_with_foreign;
+    if (inputs) {
+      inputs
+        .filter(input => input.advanced)
+        .forEach(input => {
+          templateValues[input.name] = input?.default || '';
+        });
+    }
     setAdvancedValues({
+      ...advancedValues,
```

Comment:
can the `setAdvancedValues` state will be managed only inside the component,
Looks like `AdvancedFields` could manage itself, without needing to update the parent, and if I am not wrong,
seems that no one else uses this state in the parent.

Reply:
No because the parent component will send the advanced value to the server on submit

Reply:
> Please notice that `Effective User` and `Timeout to kill` are supposed to be reset because different job templates have different defaults. If its confusing for the user I can reset all the values.

From debugging it, I see added a few console.logs
and I see that the Advanced step gets remounted, e.g:

- changed `Effective User` and `Timeout to kill`,
`// advancedValues = {"effectiveUserValue":"admin","timeoutToKill":"12","templateValues":{}}`
- changing job category
`// in did mount, advancedValues = {}`
`// after useEffect advancedValues = {"effectiveUserValue":"","timeoutToKill":"","templateValues":{"pre_script":"","post_script":""}}`

so I don't think you can retrieve the old values in `advancedValues`

Reply:
I changed the set default function to be 
``` js
    setAdvancedValues(currentAdvancedValues => ({
      ...currentAdvancedValues,
      effectiveUserValue:
        responseJob.effective_user?.value ||
        currentAdvancedValues.effectiveUserValue ||
        '',
 ```
 And after changing job templates or category the old effective user was still there